airbyte-db/lib/src/main/java/io/airbyte/db/check/DatabaseAvailabilityCheck.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright (c) 2021 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.db.check;
+
+import static org.jooq.impl.DSL.select;
+
+import io.airbyte.commons.lang.Exceptions;
+import io.airbyte.db.Database;
+import java.util.Optional;
+import java.util.function.Function;
+import org.jooq.DSLContext;
+import org.slf4j.Logger;
+
+/**
+ * Performs a check to verify that the configured database is available.
+ */
+public interface DatabaseAvailabilityCheck extends DatabaseCheck {
+
+  /**
+   * The number of times to check if the database is available. TODO replace with a default value in a
+   * value injection annotation
+   */
+  int NUM_POLL_TIMES = 10;
+
+  /**
+   * Checks whether the configured database is available.
+   *
+   * @throws InterruptedException if unable to perform the check.
+   */
+  default void check() throws InterruptedException {
+    var initialized = false;
+    var totalTime = 0;
+    final var sleepTime = getTimeoutMs() / NUM_POLL_TIMES;
+
+    while (!initialized) {
+      getLogger().warn("Waiting for database to become available...");
+      if (totalTime >= getTimeoutMs()) {
+        throw new InterruptedException("Unable to connect to the database.");
+      }
+
+      final Optional<DSLContext> dslContext = getDslContext();
+
+      if (dslContext.isPresent()) {
+        final Database database = new Database(dslContext.get());
+        initialized = isDatabaseConnected(getDatabaseName()).apply(database);
+        if (!initialized) {
+          getLogger().info("Database is not ready yet. Please wait a moment, it might still be initializing...");
+          Exceptions.toRuntime(() -> Thread.sleep(sleepTime));
```

Comment:
Why do we need to convert exceptions to Runtime here if this method already throws `InterruptedException`?

Reply:
The original code did not throw `InterruptedException`.  I did that as part of the conversion to avoid the unseen runtime exceptions being thrown.  To your point, I can remove the use of the `Exceptions.toRuntime()` call in the new code.  Will update.